hack/test-integration.sh is flaky, running too close to timeout

```
W0925 06:58:17.217146   10376 master.go:267] Network range for service cluster IPs is unspecified. Defaulting to 10.0.0.0/24.
I0925 06:58:17.217220   10376 master.go:293] Node port range unspecified. Defaulting to 30000-32767.
I0925 06:58:17.217690   10376 master.go:315] Will report 172.17.15.90 as public IP address.
E0925 06:58:25.411537   10376 handlers.go:37] Unable to authenticate the request due to an error: Token has been invalidated
E0925 06:58:25.813456   10376 handlers.go:37] Unable to authenticate the request due to an error: Token has been invalidated
E0925 06:58:26.244807   10376 handlers.go:37] Unable to authenticate the request due to an error: Token has been invalidated
E0925 06:58:26.609834   10376 handlers.go:37] Unable to authenticate the request due to an error: Token has been invalidated
panic: test timed out after 2m0s

goroutine 2729 [running]:
runtime.panic(0xed2200, 0xc20a429600)
	/usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
testing.func·008()
	/usr/local/go/src/pkg/testing/testing.go:629 +0xe6
created by time.goFunc
	/usr/local/go/src/pkg/time/sleep.go:121 +0x47

goroutine 16 [chan receive]:
testing.RunTests(0x1869148, 0x20a3960, 0x1b, 0x1b, 0x1)
	/usr/local/go/src/pkg/testing/testing.go:505 +0x923
testing.Main(0x1869148, 0x20a3960, 0x1b, 0x1b, 0x21dd360, 0x0, 0x0, 0x21dd360, 0x0, 0x0)
	/usr/local/go/src/pkg/testing/testing.go:435 +0x84
main.main()
	k8s.io/kubernetes/test/integration/_test/_testmain.go:99 +0x9c

goroutine 19 [finalizer wait, 2 minutes]:
runtime.park(0x414760, 0x21d0668, 0x20a9c29)
	/usr/local/go/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0x21d0668, 0x20a9c29)
	/usr/local/go/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
	/usr/local/go/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
	/usr/local/go/src/pkg/runtime/proc.c:1445

goroutine 22 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x21d67e0)
	/root/workspace/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x75
created by github.com/golang/glog.init·1
	/root/workspace/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2b2

goroutine 17 [syscall, 2 minutes]:
runtime.goexit()
	/usr/local/go/src/pkg/runtime/proc.c:1445

goroutine 29 [syscall, 1 minutes]:
os/signal.loop()
	/usr/local/go/src/pkg/os/signal/signal_unix.go:21 +0x1e
	/root/workspace/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/master/controller.go:90 +0x30
created by k8s.io/kubernetes/pkg/util.(*Runner).Start
	/root/workspace/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/runner.go:45 +0x112

goroutine 42 [IO wait, 1 minutes]:
net.runtime_pollWait(0x7fa086720878, 0x72, 0x0)
	/usr/local/go/src/pkg/runtime/netpoll.goc:146 +0x66
net.(*pollDesc).Wait(0xc20823d5d0, 0x72, 0x0, 0x0)
	/usr/local/go/src/pkg/net/fd_poll_runtime.go:84 +0x46
net.(*pollDesc).WaitRead(0xc20823d5d0, 0x0, 0x0)
	/usr/local/go/src/pkg/net/fd_poll_runtime.go:89 +0x42
net.(*netFD).Read(0xc20823d570, 0xc20828c000, 0x1000, 0x1000, 0x0, 0x7fa086718418, 0xb)
	/usr/local/go/src/pkg/net/fd_unix.go:232 +0x34c
net.(*conn).Read(0xc2080447a8, 0xc20828c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/pkg/net/net.go:122 +0xe7
net/http.noteEOFReader.Read(0x7fa086720950, 0xc2080447a8, 0xc2082690d8, 0xc20828c000, 0x1000, 0x1000, 0x21ebc60, 0x0, 0x0)
	/usr/local/go/src/pkg/net/http/transport.go:1203 +0x72
net/http.(*noteEOFReader).Read(0xc208284260, 0xc20828c000, 0x1000, 0x1000, 0xc208288050, 0x0, 0x0)
	:124 +0xca
bufio.(*Reader).fill(0xc20828a000)
	/usr/local/go/src/pkg/bufio/bufio.go:97 +0x1b3
bufio.(*Reader).Peek(0xc20828a000, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/pkg/bufio/bufio.go:132 +0x101
net/http.(*persistConn).readLoop(0xc208269080)
	/usr/local/go/src/pkg/net/http/transport.go:782 +0x95
created by net/http.(*Transport).dialConn
	/usr/local/go/src/pkg/net/http/transport.go:600 +0x93f

goroutine 43 [select, 1 minutes]:
net/http.(*persistConn).writeLoop(0xc208269080)
	/usr/local/go/src/pkg/net/http/transport.go:885 +0x38f
created by net/http.(*Transport).dialConn
	/usr/local/go/src/pkg/net/http/transport.go:601 +0x957
```

https://app.shippable.com/builds/5604defed7130e0b00462d60

The first line logged from the integration test is
```
I0925 06:56:31.912444   10376 etcd_utils.go:64] Deleting all etcd keys
```
the last line from the service account token test is
```
E0925 06:58:26.609834   10376 handlers.go:37] Unable to authenticate the request due to an error: Token has been invalidated
```
then the test panics with:
```
panic: test timed out after 2m0s
```

I think the tests are just running too close the the timeout. On successful runs, the tests take ~115 seconds, which is ridiculously close the the threshold

--- a/hack/test-integration.sh
+++ b/hack/test-integration.sh
@@ -31,6 +31,8 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 # KUBE_TEST_API_VERSIONS=${KUBE_TEST_API_VERSIONS:-"v1,experimental/v1alpha1"}
 KUBE_TEST_API_VERSIONS=${KUBE_TEST_API_VERSIONS:-"v1,experimental/v1alpha1"}
 
+# Give integration tests longer to run
+KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 240s}
 KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=${KUBE_INTEGRATION_TEST_MAX_CONCURRENCY:-"-1"}
 LOG_LEVEL=${LOG_LEVEL:-2}
 
@@ -48,6 +50,7 @@ runTests() {
   # KUBE_RACE="-race"
   KUBE_GOFLAGS="-tags 'integration no-docker' " \
     KUBE_RACE="" \
+    KUBE_TIMEOUT="${KUBE_TIMEOUT}" \
     KUBE_TEST_API_VERSIONS="$1" \
     KUBE_API_VERSIONS="v1,experimental/v1alpha1" \
     "${KUBE_ROOT}/hack/test-go.sh" test/integration

